### DISCUSSION TITLE
How can i add html or body classes from any page?

### DISCUSSION THREAD

**Discussion by niklasgrewe** [remix-run/react-router#8971] [Q&A]
Hi, i would like to find a way to dynamically add html or body classes from any page. Is there a way to do this out of box? In next.js you could expose `pageProps` to do it like this:

```js
// src/pages/_document.js

import Document, { Html, Head, Main, NextScript } from 'next/document';

export default class MyDocument extends Document {
  render() {
    const pageProps = this.props?.__NEXT_DATA__?.props?.pageProps;
    return (
      <Html>
        <Head />
        <body className={pageProps.isDark ? 'dark-mode' : 'light-mode'}>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}
```
I know that i can import different `.css` files per page like `light.css` or `dark.css` - but i use tailwindcss, so i need a way to add the `dark` class to the html element. But how can i do that per page? Yes, i can also create a `<Layout />` component and wrap everthing inside it, but doing this way, i need to add the layout component on every page. 

Are there other/better solutions? How do you do it?

**Comment by schizek** [comment]
I'm looking to programmatically add a class to the `body` tag as well.  Were you able to figure this out?

**Comment by Dobby89** [comment]
I'm also wanting to do this :)

**Comment by coreyleelarson** [comment]
I'm accomplishing this with a `handle` in the route and `useMatches` in the root:

**routes/admin.tsx**
```tsx
import styles from "~/styles/routes/AdminLayout.module.css";

export const handle = {
  bodyClass: styles.body,
};
```

**root.tsx**
```tsx
import { type UIMatch, useMatches } from "@remix-run/react";
import clsx from "clsx";

export default function App() {
  const matches = useMatches() as UIMatch<unknown, { bodyClass: string }>[];
  const routeBodyClasses = matches
    .filter((match) => match.handle?.bodyClass)
    .map((match) => match.handle?.bodyClass);

  return (
    <html>
      <body className={clsx(bodyClasses)}>
        ...
      </body>
    </html>
  );
}
```
